File: src/test/java/com/merricklabs/portlandbus/PortlandBusTestModule.java
package com.merricklabs.portlandbus;

import com.google.inject.AbstractModule;
import com.merricklabs.portlandbus.external.trimet.TrimetClient;
import com.merricklabs.portlandbus.mocks.MockMyStopStorage;
import com.merricklabs.portlandbus.mocks.MockTrimetClient;
import com.merricklabs.portlandbus.storage.MyStopStorage;

public class PortlandBusTestModule extends AbstractModule {

    @Override
    protected void configure(){
        bind(TrimetClient.class).to(MockTrimetClient.class);
        bind(MyStopStorage.class).to(MockMyStopStorage.class);
    }
}

File: src/test/kotlin/com/merricklabs/portlandbus/PortlandBusTestModule.kt
package com.merricklabs.portlandbus

import com.merricklabs.portlandbus.external.trimet.TrimetClient
import com.merricklabs.portlandbus.handlers.CancelandStopIntentHandler
import com.merricklabs.portlandbus.handlers.FallbackIntentHandler
import com.merricklabs.portlandbus.handlers.GetArrivalsIntentHandler
import com.merricklabs.portlandbus.handlers.HelpIntentHandler
import com.merricklabs.portlandbus.handlers.LaunchRequestHandler
import com.merricklabs.portlandbus.handlers.MyStopIntentHandler
import com.merricklabs.portlandbus.handlers.SaveStopIntentHandler
import com.merricklabs.portlandbus.handlers.SessionEndedRequestHandler
import com.merricklabs.portlandbus.mocks.MockMyStopStorage
import com.merricklabs.portlandbus.mocks.MockTrimetClient
import com.merricklabs.portlandbus.storage.MyStopStorage
import com.merricklabs.portlandbus.util.SkillsHelper
import org.koin.dsl.module.module

val PortlandBusTestModule = module {
    single { PortlandBusConfig() }
    single { SkillsHelper() }

    single { MockTrimetClient() as TrimetClient }
    single { MockMyStopStorage() as MyStopStorage }

    single { CancelandStopIntentHandler() }
    single { FallbackIntentHandler() }
    single { GetArrivalsIntentHandler() }
    single { HelpIntentHandler() }
    single { LaunchRequestHandler() }
    single { MyStopIntentHandler() }
    single { SaveStopIntentHandler() }
    single { SessionEndedRequestHandler() }
}